src/base/RegexPattern.h

```diff
@@ -26,33 +24,28 @@ class RegexPattern
 
 public:
     RegexPattern() = delete;
-    RegexPattern(const SBuf &aPattern, int aFlags);
-    ~RegexPattern();
-
     RegexPattern(RegexPattern &&) = delete; // no copying of any kind
+    RegexPattern(const SBuf &aPattern, std::regex::flag_type);
```

Comment:
I would promise to add certain flags, so that the callers do not have to bother and can rely on us doing that:

```suggestion
    /// std::regex::nosubs and std::regex::optimize flags are added automatically
    RegexPattern(const SBuf &aPattern, std::regex::flag_type);
```